src/components/Launcher.js

```diff
@@ -10,33 +10,51 @@ class Launcher extends Component {
   constructor() {
     super();
     this.state = {
-      active: false,
-      launcherIcon
+      launcherIcon,
+      isOpen: false
     };
   }
 
+  displayNewMessagesCount() {
+    if (this.props.newMessagesCount !== 0 && this.state.isOpen === false) {
+      return (
+        <div className={"sc-new-messsages-count"}>
+          {this.props.newMessagesCount}
+        </div>
+      )
+    } 
+  }
+
   handleClick() {
-    this.setState({
-      active: !this.state.active
-    });
+    if (this.props.handleClick !== undefined) {
+      this.props.handleClick();
+    } else {
+      this.setState({
+        isOpen: !this.state.isOpen,
+      });
+    }
   }
 
   render() {
+    const isOpen = (this.props.isOpen === undefined) ? this.state.isOpen : this.props.isOpen;
```

Comment:
I think it's better to use `'isOpen' in this.props` here.
If the user sets `<Launcher isOpen={this.state.myMomIsHome} />` and for some reason `myMomIsHome` becomes undefined it doesn't mean they don't want to control the component anymore.